### DISCUSSION TITLE
I want to generate example string for testing

### DISCUSSION THREAD

**Discussion by Luis-omega** [lark-parser/lark#1286] [Q&A] [answered]
My grammar includes indentation defined tokens (right now is `StartIndent` `IndentationSeparator` and `EndIndent` but I going to add more). 
I have look up for packages like that and found that the hypothesis package has a strategy for it but doesn't cover indentation. 

My idea is to get the already solved grammar (ready to build the parser) and use the info there to generate the shortest possible string for every production. 

I'm looking at code and for now I think that this https://github.com/lark-parser/lark/blob/7043c3ee20731ca297b93f1009e7474915d06c34/lark/load_grammar.py#L1408 is the right function to call, but I don't know, maybe I can just `Lark(grammar_as_str).grammar`? I don't know in what state the grammar is stored (after EBNF->BNF ?)

**Comment by MegaIng** [comment, marked as answer]
I would suggest you read the hypotheses implementation, it's IMO pretty understandable.

You can also look at `hyposmith`, which uses hypothesis.lark to implement the python grammar including the indentation.

**Comment by Luis-omega** [comment]
Right, I forgot they should have a example of how to extract the grammar. 

also, I going to check hyposmith, sounds like I was looking for it! thanks!

**Comment by Luis-omega** [comment]
At the end I follow the hypotesis script and modify the state to be a `Document` (in the same sense as in a pretty printer) , and now I have a powerfull tool to help me debug my grammar! 

Thanks for the help.
